larky/src/main/java/com/verygood/security/larky/modules/vgs/metrics/spi/LarkyMetrics.java

```diff
@@ -0,0 +1,11 @@
+package com.verygood.security.larky.modules.vgs.metrics.spi;
+
+import net.starlark.java.eval.Dict;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.StarlarkValue;
+
+public interface LarkyMetrics extends StarlarkValue {
+
+  void track(Dict<String, Object> dict) throws EvalException;
```

Comment:
Do we define somewhere what keys can be in this `Dict`? I'm kind of looking at this and thinking that it doesn't give much guidance to the user. 
If we're logging only certain keys then maybe we should not use `Dict` and just have a bunch of optional parameters.

Reply:
Sample Script : [atlas/horizon/service/grpc/transaction_metrics.star](https://github.com/verygoodsecurity/atlas-horizon/pull/385/files#diff-5106a2e5a3d77db2d8b85140b3bde69aa5b08e42e74131e07acd5fa8c1e32efe)

Keys Allowed : [atlas/commons/larky/metrics/TransactionMetricsLogsAppender.java](https://github.com/verygoodsecurity/atlas-commons/pull/198/files#diff-61cabba0d67c13f42a27ac81a4fdd653b8c0e58e5004743f0f7c7856871f3969)

Reply:
Moved `TransactionMetricsKeys.java` file to `starlarky` instead of `atlas-commons`

Reply:
We have used `Dictionary` as in future the list of **allowed keys will increase** and we then will not have to change the signature for that.

Reply:
What's wrong with changing the signature?

Reply:
signature changed